frame/contracts/src/exec.rs

```diff
@@ -158,6 +158,15 @@ pub trait Ext: sealing::Sealed {
 	/// Returns a reference to the account id of the caller.
 	fn caller(&self) -> &AccountIdOf<Self::T>;
 
+	/// Check if a contract lives at the specified `address`.
+	fn is_contract(&self, address: AccountIdOf<Self::T>) -> bool;
```

Comment:
We should be able to take a reference here instead

Reply:
hmm really
@athei @cmichi please confirm (as it differs from the function signature given in the [original issue](https://github.com/paritytech/substrate/issues/9064))

Reply:
I wouldn't necessarily consider the original issue the be-all-end-all. In this case the implementation of `is_contract` is using a `&address` internally anyways, so why not give the caller flexibility and allow them to pass a reference? See the [Rust API guidelines](https://rust-lang.github.io/api-guidelines/flexibility.html#caller-decides-where-to-copy-and-place-data-c-caller-control) for this as well

Reply:
sure, fair enough

Reply:
Yes I agree. Should be reference. Was an oversight in the original issue. Those are usually heavily under specified.